src/services/motion/client.test.ts

```diff
@@ -0,0 +1,509 @@
+        expectedObstaclesList,
+        expectedMotionConfiguration,
+        expectedExtra
+      )
+    ).resolves.toStrictEqual(testExecutionId);
+    expect(mock).toHaveBeenCalledOnce();
+
+    expect(executionId).toHaveBeenCalledOnce();
+  });
+});
+
+describe('stopPlan', () => {
+  beforeEach(() => {
+    MotionServiceClient.prototype.stopPlan = vi
+      .fn()
+      .mockImplementation((_req, _md, cb) => {
+        cb(null, {});
+      });
+  });
+
+  test('return error when not provided a componentName', async () => {
+    await expect(motion.stopPlan()).rejects.toThrowError(namespaceError);
+  });
+
+  test('return success otherwise', async () => {
+    const expectedComponenetName = {
+      namespace: 'viam',
+      type: 'component',
+      subtype: 'base',
+      name: 'myBase',
+    };
+    const expectedExtra = {};
+    const mock = vi
+      .fn()
+      .mockImplementation((req: pb.StopPlanRequest, _md, cb) => {
+        expect(req.getComponentName().toObject()).toStrictEqual(
+          expectedComponenetName
+        );
+        expect(req.getExtra().toObject()).toStrictEqual(
+          Struct.fromJavaScript(expectedExtra).toObject()
+        );
+        cb(null, {});
+      });
+    MotionServiceClient.prototype.stopPlan = mock;
+    await expect(
+      motion.stopPlan(expectedComponenetName)
+    ).resolves.toStrictEqual(null);
+  });
+
+  test('allows optionally specifying extra', async () => {
+    const expectedComponenetName = {
+      namespace: 'viam',
+      type: 'component',
+      subtype: 'base',
+      name: 'myBase',
+    };
+    const expectedExtra = { some: 'extra' };
+    const mock = vi
+      .fn()
+      .mockImplementation((req: pb.StopPlanRequest, _md, cb) => {
+        expect(req.getComponentName().toObject()).toStrictEqual(
+          expectedComponenetName
+        );
+        expect(req.getExtra().toObject()).toStrictEqual(
+          Struct.fromJavaScript(expectedExtra).toObject()
+        );
+        cb(null, {});
+      });
+    MotionServiceClient.prototype.stopPlan = mock;
+    await expect(
+      motion.stopPlan(expectedComponenetName, expectedExtra)
+    ).resolves.toStrictEqual(null);
+  });
+});
+
+describe('getPlan', () => {
+  const expectedResponse = {
+    toObject: () => ({
+      currentPlanWithStatus: {
+        plan: {
+          id: 'planId',
+          componentName: {
+            namespace: 'viam',
+            type: 'component',
+            subtype: 'base',
+            name: 'myBase',
+          },
+          executionId: 'executionId',
+          stepsList: [
+            [
+              'viam:component:base/myBase',
+              {
+                pose: {
+                  x: 10,
+                  y: 20,
+                  z: 30,
+                  oX: 40,
+                  oY: 50,
+                  oZ: 60,
+                  theta: 70,
+                },
+              },
+            ],
+          ],
+        },
+        status: {
+          state: 1,
```

Comment:
@maximpertsov how do I specify that this value is `motion_pb.PlanStateMap.PLAN_STATE_IN_PROGRESS`?

Reply:
done